File: src/main/java/brightspark/landmanager/data/areas/Position.java
package brightspark.landmanager.data.areas;

import net.minecraft.nbt.NBTTagCompound;
import net.minecraft.util.math.BlockPos;
import net.minecraftforge.common.util.INBTSerializable;

public class Position implements INBTSerializable<NBTTagCompound>
{
    public int dimensionId;
    public BlockPos position;

    public Position(int dimensionId, BlockPos position)
    {
        this.dimensionId = dimensionId;
        this.position = position;
    }

    public Position(NBTTagCompound nbt)
    {
        deserializeNBT(nbt);
    }

    @Override
    public NBTTagCompound serializeNBT()
    {
        NBTTagCompound nbt = new NBTTagCompound();
        nbt.setInteger("dimension", dimensionId);
        nbt.setLong("position", position.toLong());
        return nbt;
    }

    @Override
    public void deserializeNBT(NBTTagCompound nbt)
    {
        dimensionId = nbt.getInteger("dimension");
        position = BlockPos.fromLong(nbt.getLong("position"));
    }
}

File: src/main/kotlin/brightspark/landmanager/data/areas/Position.kt
package brightspark.landmanager.data.areas

import net.minecraft.nbt.CompoundNBT
import net.minecraft.util.ResourceLocation
import net.minecraft.util.math.BlockPos
import net.minecraftforge.common.util.INBTSerializable

class Position : INBTSerializable<CompoundNBT> {
	lateinit var dimension: ResourceLocation
	lateinit var position: BlockPos

	constructor(dimension: ResourceLocation, position: BlockPos) {
		this.dimension = dimension
		this.position = position
	}

	constructor(nbt: CompoundNBT) {
		deserializeNBT(nbt)
	}

	override fun serializeNBT() = CompoundNBT().apply {
		putString("dimension", dimension.toString())
		putLong("position", position.toLong())
	}

	override fun deserializeNBT(nbt: CompoundNBT) {
		dimension = ResourceLocation(nbt.getString("dimension"))
		position = BlockPos.fromLong(nbt.getLong("position"))
	}
}
